src/libecalc/common/priority_optimizer.py

```diff
@@ -18,21 +18,68 @@
 TResult = TypeVar("TResult")
 TPriority = TypeVar("TPriority")
 
+ComponentID = str
+
 
 @dataclass
 class PriorityOptimizerResult(Generic[TResult]):
     priorities_used: TimeSeriesString
-    priority_results: Dict[datetime, Dict[PriorityID, Dict[str, TResult]]]
+    priority_results: List[typing.Any]  # TODO: typing. This is the consumer results merged based on priorities used
 
 
 @dataclass
 class EvaluatorResult(Generic[TResult]):
-    id: str
+    id: ComponentID
     result: TResult
     is_valid: TimeSeriesBoolean
 
 
 class PriorityOptimizer(Generic[TResult, TPriority]):
+    @staticmethod
+    def get_component_ids(
+        priority_results: Dict[datetime, Dict[PriorityID, Dict[ComponentID, TResult]]]
```

Comment:
not a TemporalModel? Should create a new class for this (soon?), as it gets (too) many layers of dicts ...difficult to grasp